libethereum/Client.cpp

```diff
@@ -508,7 +508,14 @@ void Client::restartMining()
     // TODO: use m_postSeal to avoid re-evaluating our own blocks.
     preChanged = newPreMine.sync(bc());
 
-    if (preChanged || m_postSeal.author() != m_preSeal.author())
+    Address preAuthor;
+    DEV_READ_GUARDED(x_preSeal)
+        preAuthor = m_preSeal.author();
+    Address postAuthor;
+    DEV_READ_GUARDED(x_postSeal)
+        postAuthor = m_postSeal.author();
+
+    if (preChanged || postAuthor != preAuthor)
```

Comment:
This seems to be still wrong. You are making a decision based on the values that might be not longer valid.

Reply:
```
guard(x_preseal)
     guard(x_postSeal)
          condition = postAuthor != preAuthor
endguards

if (preChanged ||  condition) 
{
    function body
}
```
I cant move function body under the guard because it will make locks recursive. 
is there other way to make guards?